### PR TITLE
save a batch file in the backup for convenience

### PR DESCRIPTION
**Pull request by MaximilianMeister**
this could be used during an upgrade if a user chooses to apply all
barclamps at once

depends on https://github.com/crowbar/crowbar-core/pull/477

### CODE REVIEW

File: crowbar_framework/app/models/batch/export.rb [hunk 1]
```diff
@@ -17,6 +17,8 @@
 
 module Batch
   class Export < Base
+    attr_accessor :path
```
Comment (rsalevsky) [inline crowbar_framework/app/models/batch/export.rb]: Why is this needed?

Comment (MaximilianMeister) [inline crowbar_framework/app/models/batch/export.rb]: otherwise I wouldn't be able to set `path` which means I cannot set a custom path for the yml output (the backup path in our case)

Comment (rsalevsky) [inline crowbar_framework/app/models/batch/export.rb]: Ah missed this.

**Comment by rsalevsky** [comment]
+1

File: crowbar_framework/app/models/batch/export.rb (hunk 1 above)
Comment (MaximilianMeister) [inline crowbar_framework/app/models/batch/export.rb]: but now that you mention it, shall I also make filename accessible? it is not used yet though...

Comment (rsalevsky) [inline crowbar_framework/app/models/batch/export.rb]: If you want to use a custom file name then yes, otherwise it makes not really sense if we don't use it.

File: crowbar_framework/lib/crowbar/backup/export.rb [hunk 2]
```diff
@@ -128,6 +129,15 @@ def crowbar
         end
       end
 
+      def batch
+        logger.debug "Backing up batch export"
```
Comment (toabctl) [inline crowbar_framework/lib/crowbar/backup/export.rb]: can you add the path to the backup file to the log?

Comment (MaximilianMeister) [inline crowbar_framework/lib/crowbar/backup/export.rb]: doesn't make sense here as this is just a substep of the backup creation. and logging the path is already done at the beginning of the backup creation in https://github.com/crowbar/crowbar-core/blob/master/crowbar_framework/app/models/backup.rb#L160

**Comment by toabctl** [comment]
+1

**Comment by nicolasbock** [comment]
+1

**Comment by MaximilianMeister** [comment]
https://github.com/crowbar/crowbar-core/issues/477 is merged now, so we can merge this too
